lib/document-builder.ts

```diff
@@ -110,74 +115,147 @@ export class DocumentBuilder {
   }
 
   public addBearerAuth(
-    options: SecuritySchemeObject = {
-      type: 'http'
-    },
-    name = 'bearer'
+    name?: string,
+    options?: Partial<Optional<HttpSchemeObject, 'type'>>
+  ): this;
+  /**
+   * @deprecated Use `addBearerAuth(name, options)` instead
+   */
+  public addBearerAuth(
+    options?: Partial<Optional<HttpSchemeObject, 'type'>>,
+    name?: string
+  ): this;
+  public addBearerAuth(
+    nameOrOptions:
+      | string
+      | Partial<Optional<HttpSchemeObject, 'type'>> = 'bearer',
+    optionsOrName: string | Partial<Optional<HttpSchemeObject, 'type'>> = {}
   ): this {
-    this.addSecurity(name, {
+    if (typeof nameOrOptions === 'object') {
+      [nameOrOptions, optionsOrName] = [optionsOrName, nameOrOptions];
+    }
+    this.addSecurity(nameOrOptions as string, {
+      type: 'http',
       scheme: 'bearer',
       bearerFormat: 'JWT',
-      ...options
+      ...(optionsOrName as any)
     });
     return this;
   }
 
   public addOAuth2(
-    options: SecuritySchemeObject = {
-      type: 'oauth2'
-    },
-    name = 'oauth2'
+    name?: string,
+    options?: Partial<Optional<OAuth2SchemeObject, 'type'>>
+  ): this;
+  /**
+   * @deprecated Use `addOAuth2(name, options)` instead
+   */
+  public addOAuth2(
+    options?: Partial<Optional<OAuth2SchemeObject, 'type'>>,
+    name?: string
+  ): this;
+  public addOAuth2(
+    nameOrOptions:
+      | string
+      | Partial<Optional<OAuth2SchemeObject, 'type'>> = 'oauth2',
+    optionsOrName: string | Partial<Optional<OAuth2SchemeObject, 'type'>> = {}
   ): this {
-    this.addSecurity(name, {
+    if (typeof nameOrOptions === 'object') {
+      [nameOrOptions, optionsOrName] = [optionsOrName, nameOrOptions];
+    }
+    this.addSecurity(nameOrOptions as string, {
       type: 'oauth2',
       flows: {},
-      ...options
+      ...(optionsOrName as any)
```

Comment:
I guess we need to change the signature of `addSecurity` instead of using type assertion here and in other places, right?

Reply:
@micalevisk, sorry, I still don't understand what you meant. 😨

Reply:
I mean, can't we avoid the `as any` here?

Reply:
@micalevisk I'm sorry, but I really don't have time to improve it. I have already tried to update the typing so that it is more correct for developers outside the module.
And this part that you are asking to fix will not be so important outside, but only inside this module. This can be marked as TODO at least and make these changes later.

Reply:
it's fine

```suggestion
      // TODO: avoid using type assertion to 'any' below
      ...(optionsOrName as any)
```